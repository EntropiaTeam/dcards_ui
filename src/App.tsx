import React, { FC } from 'react';
import { ThemeProvider } from '@material-ui/core/styles';
import { GlobalProvider } from './context/GlobalProvider';
import { InitialDataProvider } from './context/initialData';
import { Main } from './containers';
import mainTheme from './config/theme';
import AdobeAnalytics from './utils/AdobeAnalytics';
import TelemtryProvider from './utils/TelemetryProvider';
import './AppFonts.css';

const App: FC = () => {
  if (window.APP_INSIGHTS_INSTRUMENTATION_KEY) {
    TelemtryProvider.init(window.APP_INSIGHTS_INSTRUMENTATION_KEY);
  } else {
    // eslint-disable-next-line no-console
    console.log('Fault: Telemetry fallback to PRD.');
    TelemtryProvider.init('52f7878a-ed5d-4c41-add6-af83ce6315f3');
  }
  AdobeAnalytics.getInstance().pageInit();

  return (
    <GlobalProvider>
      <InitialDataProvider>
        <ThemeProvider theme={mainTheme}>
          <Main />
        </ThemeProvider>
      </InitialDataProvider>
    </GlobalProvider>
  );
};

export default App;
