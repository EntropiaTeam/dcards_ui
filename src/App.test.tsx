import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('test App layout', () => {
  const AppComponent = renderer.create(<App />).toJSON();

  it('should render App Component correctly', () => {
    expect(AppComponent).toMatchSnapshot();
  });
});
