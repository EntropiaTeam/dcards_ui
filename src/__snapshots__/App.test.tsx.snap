// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App layout should render App Component correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-6"
>
  <div
    className="MuiBox-root MuiBox-root-7"
  >
    <img
      alt="edible-logo"
      src="https://resources.ediblearrangements.com/resources/en-us/i/revamp_p/edible_arrangement_logo.png?v=09302019"
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-8"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      Page not found
    </h6>
  </div>
</div>
`;
